presto-docs/src/main/sphinx/functions/aggregate.rst

```diff
@@ -105,6 +105,15 @@ General Aggregate Functions
 
     Returns ``n`` smallest values of all input values of ``x``.
 
+.. function:: reduce_agg(inputValue T, initialState S, inputFunction(S, T, S), combineFunction(S, S, S)) -> S
+
+    Returns a single value reduced from input values. ``inputFunction`` and ``combineFunction`` will be invoked.
+    ``inputFunction`` takes the current state (initially ``initialState``) and the input value, and returns the new state.
+    ``combineFunction`` takes two states and returns the new state.
+    The final state is returned.
+
+    ``redcue_agg`` supports state with native container type long, double or boolean.
```

Comment:
does it mean that `S` must be one of these?

(just curious, why so?)

Reply:
typo: redcue_agg -> reduce_agg

Could you add an example of usage?

Reply:
@findepi : Correct, see the comments in code: https://github.com/prestodb/presto/blob/79f480bc807954c4486b670c42015fb9c396f010/presto-main/src/main/java/com/facebook/presto/operator/aggregation/ReduceAggregationFunction.java#L124-L127

Also see https://github.com/prestodb/presto/issues/9553